src/main/webapp/login/login.controller.js

```diff
@@ -1,5 +1,29 @@
 'use strict';
 
+var ispn = ispn || {};
```

Comment:
This is "wild" javascript code: it is global, and we should not write global code when using angular. So in this case we want to expose some util library, so it can be used by other components: you should make it a service.